### DISCUSSION TITLE
SIP headers short form

### DISCUSSION THREAD

**Discussion by vak-oskh** [resiprocate/resiprocate#258] [Ideas]
Hello.
[RFC3261 says](https://www.rfc-editor.org/rfc/rfc3261.html#section-7.3.3) that SIP protocol provides a short form of header field names.
I've found that reciprocate supports parsing of the short headers form ([resip/stack/HeaderHash.gperf](https://github.com/resiprocate/resiprocate/blob/master/resip/stack/HeaderHash.gperf)). After parsing, the library initializes SIP headers with a regular form and use them for further processing and sending.
But it is not possible to create SIP message with the short headers form to send to a wire.

Did you make plans to add this feature in the next releases?

**Comment by oej** [comment]
General answer: This was created 20 years ago when bandwidth was a concern. Today the biggest concern is the size of the SDP. No new headers have short forms so the impact of using short form headers in SIP will be minimal.

**Reply by vak-oskh** [reply]
The main idea is clear. Thanks for the feedback.

**Comment by sgodin** [comment]
Note:  at one point in time a company using resip added SigComp support for making SIP messaging smaller.  I'm not aware of anyone still using it, or at least maintaining it's support in the stack.  It may still work.  I believe you need a enable compile time flag for it, and I seem to remember there were issues trying to locate the OpenSigComp library.  :)

https://datatracker.ietf.org/doc/html/rfc5049

**Reply by vak-oskh** [reply]
Thank you for the tip. I will see if this could be applied to my case.
